File: sources/contract-registry/src/main/java/fr/unice/polytech/si5/al/e/contractRegistry/interfaces/HandleContract.java
package fr.unice.polytech.si5.al.e.contractRegistry.interfaces;

import fr.unice.polytech.si5.al.e.contractRegistry.exceptions.NoSuchContractIdException;
import fr.unice.polytech.si5.al.e.model.Contract;
import fr.unice.polytech.si5.al.e.model.type.Type;

import javax.ejb.Local;

@Local
public interface HandleContract {
    Contract addContract(Type type, String description, String mail);
    Contract updateContractDescription(int id,String description) throws NoSuchContractIdException;
    void clear();
}

File: blablamoveService/typescript/main/src/components/contract-registry/HandleContract.ts
import {Type} from "../../entity/Type";
import {Contract} from "../../entity/contract/Contract";
import { Policy } from "../../entity/Policy/Policy";

export interface HandleContract {

    addContract(type: Type,description: string, mail: string, policies: Array<Policy>) : Promise<Contract>
    updateContractDescription(id: number, description: string) : Promise<Contract>

}